overlord/devicestate/handlers_install.go

```diff
@@ -34,7 +41,27 @@ func (m *DeviceManager) doSetupRunSystem(t *state.Task, _ *tomb.Tomb) error {
 	perfTimings := timings.NewForTask(t)
 	defer perfTimings.Save(st)
 
-	// XXX: add handler content
+	// get gadget dir
+	deviceCtx, err := DeviceCtx(st, t, nil)
+	if err != nil {
+		return fmt.Errorf("cannot get device context: %v", err)
+	}
+	info, err := snapstate.GadgetInfo(st, deviceCtx)
+	if err != nil {
+		return fmt.Errorf("cannot get gadget info: %v", err)
+	}
+	gadgetDir := info.MountDir()
+
+	// run the create partition code
+	st.Unlock()
+	output, err := exec.Command(filepath.Join(dirs.DistroLibExecDir, "snap-bootstrap"), "create-partitions", gadgetDir).CombinedOutput()
+	st.Lock()
+	if err != nil {
+		return osutil.OutputErr(output, err)
```

Comment:
I'm wondering if it would make sense to give any extra context here in the error message, or is the raw error from snap-bootstrap enough?

Reply:
we should probably have an error case test too

Reply:
Thank you! I add a test in a followup and cleanup/improve the error as needed.